src/main/java/edu/kit/datamanager/repo/util/validators/impl/URLValidator.java

```diff
@@ -48,27 +50,33 @@ public boolean isValid(String input) {
         LOGGER.debug("URL: {}", input);
         int status;
         boolean result;
+        String regex = "^(http|https):\\/\\/[-A-Za-z0-9+&@#\\/%?=~_|!:,.;]+[-A-Za-z0-9+&@#\\/%=~_|]$";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(input);
+
+        if (!matcher.find()) {
+            LOGGER.error("The URL {} does not match the pattern or contains illegal characters.", input);
+            throw new BadArgumentException("The URL " + input + " does not match the pattern or contains illegal characters.");
+        }
 
         try {
             urlHandler = new URL(input);
             con = (HttpURLConnection) urlHandler.openConnection();
             con.setRequestMethod("GET");
             status = con.getResponseCode();
             LOGGER.debug("HTTP status: {}", status);
-//            if (status == HttpStatus.SC_OK || status == HttpStatus.SC_MOVED_TEMPORARILY || status == HttpStatus.SC_TEMPORARY_REDIRECT) {
-            if(status == HttpStatus.SC_OK) {
+            if (status == HttpStatus.SC_OK) {
                 result = true;
             } else {
                 LOGGER.error("Invalid URL");
```

Comment:
Please provide more information.
Which URL seems to be invalid and maybe why.
You may use a switch command with different status codes.
Or use a generalized message: Connection to URL '{}' fails with status code: '{}'